ios/Classes/core/PMManager.m

```diff
@@ -1092,6 +1092,36 @@ - (void)saveVideo:(NSString *)path
     }];
 }
 
+- (void)saveLivePhoto:(NSString *)imagePath
+            videoPath:(NSString *)videoPath
+            title:(NSString *)title
+             desc:(NSString *)desc
+            block:(AssetResult)block {
+    [PMLogUtils.sharedInstance info:[NSString stringWithFormat:@"save LivePhoto with imagePath: %@, videoPath: %@, title: %@, desc %@",
+                                     imagePath, videoPath, title, desc]];
+    NSURL *fileURL = [NSURL fileURLWithPath:videoPath];
+    NSURL *imageURL = [NSURL fileURLWithPath:imagePath];
+    __block NSString *assetId = nil;
+    [[PHPhotoLibrary sharedPhotoLibrary]
+     performChanges:^{
+        PHAssetCreationRequest *request = [PHAssetCreationRequest creationRequestForAsset];
+        PHAssetResourceCreationOptions *options = [PHAssetResourceCreationOptions new];
+        [options setOriginalFilename:title];
+        [request addResourceWithType:PHAssetResourceTypePhoto fileURL:imageURL options:options];
+        [request addResourceWithType:PHAssetResourceTypePairedVideo fileURL:fileURL options:options];
+        assetId = request.placeholderForCreatedAsset.localIdentifier;
+    }
+     completionHandler:^(BOOL success, NSError *error) {
```

Comment:
Did you make sure that the corresponding asset has a valid subtype to indicate as a live-photo?

Reply:
Yes, the native apple photos app also detected it as Live Photo.